activerecord/lib/active_record/associations/join_dependency.rb

```diff
@@ -92,7 +92,8 @@ def self.walk_tree(associations, hash)
       #    associations # => [:appointments]
       #    joins # =>  []
       #
-      def initialize(base, associations, joins)
+      def initialize(base, associations, joins, finder_methods_hash = {})
```

Comment:
I am not sure if we can change this API, I think this is public, @rafaelfranca ?